Emby.Server.Implementations/Library/Resolvers/Audio/MusicArtistResolver.cs

```diff
@@ -94,7 +95,18 @@ protected override MusicArtist Resolve(ItemResolveArgs args)
             var albumResolver = new MusicAlbumResolver(_logger, _fileSystem, _libraryManager);
 
             // If we contain an album assume we are an artist folder
-            return args.FileSystemChildren.Where(i => i.IsDirectory).Any(i => albumResolver.IsMusicAlbum(i.FullName, directoryService)) ? new MusicArtist() : null;
+            var directories = args.FileSystemChildren.Where(i => i.IsDirectory);
+
+            var result = Parallel.ForEach(directories, (fileSystemInfo, state) =>
+            {
+                if (albumResolver.IsMusicAlbum(fileSystemInfo.FullName, directoryService))
+                {
+                    // stop once we see a music album
+                    state.Stop();
+                }
+            });
+
+            return !result.IsCompleted ? new MusicArtist() : null;
```

Comment:
```suggestion
            return result.IsStopped ? new MusicArtist() : null;
```

I haven't used `Parallel.ForEach` in this manner before but looking at the docs it seems like checking if any iteration has called `Stop()` would be better then seeing if it completed all iterations

Reply:
It doesn't look like there is an `IsStopped` property on that object. Based on [the docs](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.parallelloopresult?view=netcore-3.1), `IsCompleted` should be good here.